poly/deal.go

```diff
@@ -234,10 +234,10 @@ type Promise struct {
  *    secretPair.Suite == longPair.Suite
  *
  * Returns
- *   A newly constructed Promise
+ *   A newly constructed Deal
  */
-func (p *Promise) ConstructPromise(secretPair *config.KeyPair,
-	longPair *config.KeyPair, t, r int, insurers []abstract.Point) *Promise {
+func (p *Deal) ConstructDeal(secretPair *config.KeyPair,
```

Comment:
I think this method should just be called Construct or Create; no need to include Deal in it since it's a Deal method.